lib/commands/xctest.js

```diff
@@ -0,0 +1,87 @@
+import B from 'bluebird';
+import { logger } from 'appium-support';
+import log from '../logger';
+import appManagement from './app-management';
+
+let helpers = {}, commands = {}, extensions = {};
+
+const xctestLog = logger.getLogger('XCTest');
+
+function assertIDB (context) {
+  if (!context.opts?.device?.idb) {
+    log.errorAndThrow(`To use XCTest runner, IDB (https://github.com/facebook/idb) must be installed ` +
+      `and sessions must be run with the "launchWithIDB" capability`);
+  }
+}
+
+/**
+ * Run an XCTest
+ *
+ * @param {object} params - The parameters that get passed to IDB to run XC test
+ *                          * testRunnerBundleId: Bundle ID of the test app
+ *                          * appUnderTestBundleId: Bundle ID of the app-under-test
+ *                          * xcTestBundleId: Bundle ID of the testrunner
+ *                          * testType: 'app', 'ui', or 'logic'
+ *                          * opts: Object containing opts.env (environment variables) and
+ *                              opts.args (test arguments)
+ */
+commands.mobileRunXCTest = async function runXCTest ({
+  testRunnerBundleId,
+  appUnderTestBundleId,
+  xctestBundleId,
+  testType = 'ui',
+  opts = {},
```

Comment:
I think there is no need for `opts` here, since args are already wrapped into an object